### PR TITLE
refactor(thumbnail): core tokens update

### PR DESCRIPTION
**Pull request by jenndiaz**
## Description

Migrates Thumbnail to the core tokens system. 

[Jira Ticket CSS-287]( https://jira.corp.adobe.com/browse/CSS-287) 

Design changes and expected VRT failures: 
- Focus has gone from 2 to 1 blue borders 
- Addition of the Layer Management and Layer + Select Variant 
- sizing changes, T-shirt to numeric 
- Adds Disabled variant 
- Border Radius has changed 
- Uses new opacity checkerboard 

## Related issue(s)

-

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/thumbnail/README.md [hunk 1]
```diff
@@ -26,69 +26,167 @@ import { Thumbnail } from '@spectrum-web-components/thumbnail';
...
+    <div class="image-wrapper">
+        <img
+            class="image"
+            style="max-block-size: 100%;
+                max-inline-size: 100%;
+                position: relative;
+                z-index: 1;"
+            src="https://place.dog/100/100"
+            alt="Demo Image"
+        />
```
Comment (jenndiaz) [inline packages/thumbnail/README.md:104]: This should work without needing the inline styling and instead should some from `class="image"`

File: packages/thumbnail/README.md [hunk 2]
```diff
@@ -26,69 +26,167 @@ import { Thumbnail } from '@spectrum-web-components/thumbnail';
...
+    <img src="https://place.dog/100/100" alt="Demo Image" />
+</sp-thumbnail>
+```
+
+</sp-tab-panel>
+<sp-tab value="500">500</sp-tab>
+<sp-tab-panel value="500">
+
+```html
+<sp-thumbnail size="500" style="display: flex;">
```
Comment (jenndiaz) [inline packages/thumbnail/README.md:94]: The implementation on spectrum-css does not need a flex on the parent div, not sure why it is needed here.

File: packages/thumbnail/README.md [hunk 3]
```diff
@@ -26,69 +26,167 @@ import { Thumbnail } from '@spectrum-web-components/thumbnail';
...
+
+```html
+<sp-thumbnail focused style="display: flex;">
+    <div class="image-wrapper">
+        <img
+            class="image"
+            style="max-block-size: 100%;
+                max-inline-size: 100%;
+                position: relative;
+                z-index: 1;"
```
Comment (jenndiaz) [inline packages/thumbnail/README.md:174]: same comments as above this should work without needing the inline styling for the ``  image or  `display: flex;`  on` <sp-thumbnail>`

File: packages/thumbnail/README.md (hunk 2 above)
Comment (Westbrook) [reply packages/thumbnail/README.md:94]: Looks like you've put the `<div class="image-wrapper">` in the wrong place. It should wrap the `<slot>` found [here](https://github.com/adobe/spectrum-web-components/pull/3137/files/b60ce804b99af52b3a26ef174fbb3bc9ed372774).

File: packages/thumbnail/src/thumbnail.css [hunk 4]
```diff
@@ -12,6 +12,6 @@ governing permissions and limitations under the License.
 
 @import './spectrum-thumbnail.css';
 
-::slotted(:not(img)) {
+/* ::slotted(:not(img)) {
```
Comment (Westbrook) [inline packages/thumbnail/src/thumbnail.css:15]: I think we will want this back after the internal structure of the element is correct to the updated implementation.

File: packages/thumbnail/src/spectrum-thumbnail.css [hunk 5]
```diff
@@ -12,353 +12,431 @@ governing permissions and limitations under the License.
...
-:host([cover]) ::slotted(*) {
-    height: 100%;
+:host([cover]) .image {
+    block-size: 100%;
+    inline-size: 100%;
     object-fit: cover;
     object-position: center;
-    width: 100%;
 }
 .background {
```
Comment (Westbrook) [inline packages/thumbnail/src/spectrum-thumbnail.css:421]: This still being available points to a feature having been removed in you edits to `Thumbnail.ts`.

File: packages/thumbnail/src/spectrum-thumbnail.css [hunk 6]
```diff
@@ -12,353 +12,431 @@ governing permissions and limitations under the License.
...
-        var(--spectrum-alias-border-color-key-focus)
+    outline-width: var(
+        --mod-thumbnail-border-width-selected,
+        var(--spectrum-thumbnail-border-width-selected)
     );
-    --spectrum-thumbnail-darksquare-background-color: var(
-        --spectrum-thumbnail-l-darksquare-background-color,
-        var(--spectrum-alias-thumbnail-darksquare-background-color)
+}
+.spectrum-Thumbnail-layer-inner {
```
Comment (Westbrook) [inline packages/thumbnail/src/spectrum-thumbnail.css:369]: This selector not having a conversion points to there being a feature in the CSS implementation that we don't leverage here. What does `layer-inner` do?

File: packages/thumbnail/src/spectrum-thumbnail.css [hunk 7]
```diff
@@ -12,353 +12,431 @@ governing permissions and limitations under the License.
...
+    border-radius: var(
+        --mod-thumbnail-border-radius,
+        var(--spectrum-thumbnail-border-radius)
     );
-    --spectrum-thumbnail-border-size: var(
-        --spectrum-thumbnail-l-border-size,
-        var(--spectrum-alias-border-size-thin)
+    overflow: hidden;
+}
+.spectrum-Thumbnail-layer {
```
Comment (Westbrook) [inline packages/thumbnail/src/spectrum-thumbnail.css:327]: Another new feature to make sure we're supporting?

File: packages/thumbnail/src/Thumbnail.ts [hunk 8]
```diff
@@ -13,23 +13,38 @@ governing permissions and limitations under the License.
...
     html,
-    SizedMixin,
     SpectrumElement,
     TemplateResult,
 } from '@spectrum-web-components/base';
 import { property } from '@spectrum-web-components/base/src/decorators.js';
 
 import styles from './thumbnail.css.js';
 
+export type ThumbnailSize =
```
Comment (Westbrook) [inline packages/thumbnail/src/Thumbnail.ts:23]: Would be good to know if design is expecting more patterns to leverage numbered t-shirt sizing. If so, we could add this as a specific feature of the `sizedMixin` to ensure that we're not doing so much copy & pasting over time.

File: packages/thumbnail/README.md (hunk 3 above)
Comment (jenndiaz) [reply packages/thumbnail/README.md]: got this fixed!

File: packages/thumbnail/src/thumbnail.css (hunk 4 above)
Comment (jenndiaz) [reply packages/thumbnail/src/thumbnail.css]: it has been put back!

File: packages/thumbnail/README.md [hunk 9]
```diff
@@ -26,72 +26,148 @@ import { Thumbnail } from '@spectrum-web-components/thumbnail';
...
-<sp-tabs selected="m" auto label="Size Attribute Options">
-<sp-tab value="xss">Extra Extra Small</sp-tab>
-<sp-tab-panel value="xxs">
+<sp-tabs selected="500" auto label="Size Attribute Options">
+<sp-tab value="50">50</sp-tab>
+<sp-tab-panel value="50">
 
 ```html
-<sp-thumbnail size="xxs">
+<sp-thumbnail size="50" style="display: block;">
```
Comment (jenndiaz) [inline packages/thumbnail/README.md:34]: We be adding this inline styling to the class for thumbnail in the spectrum-css PR and do another beta release. So all these will `style="display: block;`  be removed before the PR is ready to go.

File: packages/thumbnail/README.md (hunk 1 above)
Comment (jenndiaz) [reply packages/thumbnail/README.md]: Fixed this by using `converter.classToSlotted('spectrum-Thumbnail-image'),` in `spectrum-config.js` which added 
```
::slotted(*) {
    max-block-size: 100%;
    max-inline-size: 100%;
    position: relative;
    z-index: 1;
}
```

File: packages/thumbnail/src/spectrum-thumbnail.css (hunk 7 above)
Comment (jenndiaz) [reply packages/thumbnail/src/spectrum-thumbnail.css]: Yes this is part of a new variant, Layer which uses .spectrum-thumbnail-layer and `layer-inner` caught me mid way though getting this part implemented. It is in there now!

File: packages/thumbnail/src/spectrum-thumbnail.css (hunk 6 above)
Comment (jenndiaz) [reply packages/thumbnail/src/spectrum-thumbnail.css:369]: Code is using it now, `layer-inner` is similar to the `image-wrapper` but for the layer variant.

**Comment by jenndiaz** [comment]
@Westbrook do you know why the VRT failures and deployed link as not being commented on the PR? 

Poking around the failed VRTs in the checks tab I can see that the reviews published but  I am getting a page not found for each of the links.  
[https://e8f759e9eb6656b770b33ae44c067c85--spectrum-web-components.netlify.app](https://e8f759e9eb6656b770b33ae44c067c85--spectrum-web-components.netlify.app )

File: packages/thumbnail/README.md (hunk 9 above)
Comment (jenndiaz) [reply packages/thumbnail/README.md]: these have been removed!

**Comment by jenndiaz** [comment]
I was able to [add the new variants to the stories ](https://github.com/adobe/spectrum-web-components/pull/3137/files#diff-bd45ae627c57700c0c1ca4e26193ea377a443f213aeaf9040a173db90543c66cR34)for this component, but not the sizing. 

When attempting to update `thumbnail-sizes.storeis.ts` to follow this pattern: 
`export const 50 = (): TemplateResult => thumbnail({ size: '50' });`  instead of t-shirt sizing. 

I am getting an error when I attempt to commit the changes:
```
packages/thumbnail/stories/thumbnail-sizes.stories.ts:35:13
  ✖  35:13  Parsing error: Variable declaration expected. 
```

**Comment by Westbrook** [comment]
You can't set a number as a variable in JS. Instead of `export const 50 = (): TemplateResult => thumbnail({ size: '50' });` try something like `export const size50 = (): TemplateResult => thumbnail({ size: '50' });`...

File: packages/thumbnail/src/Thumbnail.ts (hunk 8 above)
Comment (jenndiaz) [reply packages/thumbnail/src/Thumbnail.ts:23]: I did touch base with design, it sounds like for now it is just Avatar and Thumbnail using this scale.

**Comment by pfulton** [comment]
This has been merged and released in Spectrum CSS, and the full release is: `@spectrum-css/thumbnail@3.0.0`

**Review by Westbrook — DISMISSED** [review]
Come on over, we're having a merge party!! Thanks. 🙇🏼‍♂️

**Pull request merged**